ztp/siteconfig-generator/main.go

```diff
@@ -0,0 +1,55 @@
+package main
+
+import (
+	"bytes"
+	"fmt"
+	"log"
+
+	siteConfigs "github.com/openshift-kni/cnf-features-deploy/ztp/siteconfig-generator/siteConfig"
+	"github.com/spf13/pflag"
+	"gopkg.in/yaml.v3"
+)
+
+func main() {
+	// Parse command input
+	pflag.Parse()
+
+	// Collect and parse siteconfig files paths
+	siteConfigFiles := pflag.Args()
+	var outputBuffer bytes.Buffer
+	scBuilder, _ := siteConfigs.NewSiteConfigBuilder()
+
+	for _, siteConfigFile := range siteConfigFiles {
+		fileData, err := siteConfigs.ReadFile(siteConfigFile)
+		if err != nil {
+			log.Printf("Error: could not read file %s: %s\n", siteConfigFile, err)
+			continue
```

Comment:
If any of the files cause an error, should a variable be set so that the return code at the end can be non-zero to indicate that error? I like that it continues and generates all the files but returning success at the end seems wrong if a file failed to generate. Same for other errors below.

Reply:
What about catching the error here, setting a flag and issuing the panic at the end of main()? In that way everything which can be generated successfully is done and then the panic informs kustomize that there was an issue? I don't think that argo should show "success" in the application if there are failures.

Reply:
Can you add a note here (and below for the unmarshall) to document that kustomize/argocd will handle this class of errors. This will help as we are maintaining this code to know the error case is handled.